src/screens/FreeProducts/index.tsx

```diff
@@ -0,0 +1,77 @@
+import './index.css'
+
+import React, { useState, useEffect } from 'react'
+import { Header } from 'src/components/UI'
+import { Element } from './utils/apiResponseType'
+import { extractValidPromotions, generateLink } from './utils/utils'
+import { createNewWindow } from 'src/helpers'
+import { useTranslation } from 'react-i18next'
+import { IpcRenderer } from 'electron'
+const { ipcRenderer } = window.require('electron') as {
+    ipcRenderer: IpcRenderer
+}
+
+const FreeProducts = () => {
+  const { t, i18n } = useTranslation()
+  let lang = i18n.language
+  if (i18n.language === 'pt') {
+    lang = 'pt-BR'
+  }
+  const [products, setProducts] = useState<Element[]>([])
+
+  const fetchProducts = async () => {
```

Comment:
In my opinion, we could improve the `requestFreeProducts` request to return the `validProducts` already, most of this logic would work better on the backend and the frontend should be used only for rendering that response.